gitlab/auth.go

```diff
@@ -37,6 +39,21 @@ func NewClient(token string, tokenType string, optFns ...gitprovider.ClientOptio
 		return nil, err
 	}
 
+	// Ensure that the URL includes a scheme when using a custom domain.
+	if opts.Domain != nil && *opts.Domain != "" {
+		d, err := url.Parse(*opts.Domain)
+		if err != nil {
+			return nil, err
+		}
+		if d.Scheme != "http" && d.Scheme != "https" {
```

Comment:
Isn't this covered by https://github.com/fluxcd/go-git-providers/blob/main/gitprovider/util.go#L35?

Reply:
`GetDomainURL`is used by flux to get the sync url. The gogitlab uses `setBaseURL` to setup the url. It validates the api version in the url, but not the scheme apparently.